.github/workflows/grapl-staging-release.yml

```diff
@@ -0,0 +1,108 @@
+name: Grapl Staging Release
+
+on:
+  # Every time we push to staging,
+  # release that to Dockerhub with the tag 'staging'.
+  push:
+    branches:
+    - staging
+
+jobs:
+  release-rust-services:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Install dobi
+        run: |
+          wget https://github.com/dnephin/dobi/releases/download/v0.13.0/dobi-linux
+          chmod +x dobi-linux
+
+      - name: Export tag
+        run: |
+          echo "::set-env name=TAG::staging"
+
+      - name: Build Rust services
+        run: |
+          GRAPL_RELEASE_TARGET=release ./dobi-linux --no-bind-mount rust
+
+      - name: Extract artifacts from Rust build
+        id: extract-rust-artifacts
+        run: |
+          ./etc/build_scripts/extract_rust_artifacts.sh
+
+      - name: Log in to Docker registry
+        run: |
+          echo "${{ secrets.DOCKERHUB_TOKEN }}" | docker login --username grapl --password-stdin
+
+      - name: Publish Rust images to DockerHub
+        run: |
+          ./etc/build_scripts/push_rust_to_docker.sh
+
+      - name: Clean up dist
+        run: |
+          rm -rf dist
+
+  release-python-services:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Install dobi
+        run: |
+          wget https://github.com/dnephin/dobi/releases/download/v0.13.0/dobi-linux
+          chmod +x dobi-linux
+
+      - name: Export tag
+        run: |
+          echo "::set-env name=TAG::staging"
+
+      - name: Build Python services
+        run: |
+          GRAPL_RELEASE_TARGET=release TAG="$TAG" ./dobi-linux --no-bind-mount python
```

Comment:
wait this seems dumb

Reply:
The `TAG="$TAG"` part? I mean.. I don't really have an opinion..

One question I have here is should this workflow also run the tests? I think there's a fair argument either way. On the one hand, in the majority case this workflow will run immediately after a PR merge, and the PR just ran the tests. However, the PR tests were run on the PR branch, not staging post-merge. So there's room for breakage there, potentially. Also, it's possible for an admin to force push to staging, in which case no tests would be run and we'd just blindly build & release to dockerhub.

So I'm not 100% committed to the idea that we need to run tests here but I think there are some reasons it could make sense. If we do run tests we should make sure *none* of the images are published if *any* tests fail.